powerauth-nextstep/src/main/java/io/getlime/security/powerauth/app/nextstep/controller/OperationController.java

```diff
@@ -169,4 +176,74 @@ public OperationController(OperationPersistenceService operationPersistenceServi
         return new Response<>(Response.Status.OK, responseList);
     }
 
+    /**
+     * Get all authentication methods supported by Next Step server.
+     *
+     * @return List of authentication methods wrapped in GetAuthMethodResponse.
+     */
+    @RequestMapping(value = "/auth-method", method = RequestMethod.GET)
+    public @ResponseBody
+    Response<GetAuthMethodsResponse> getAuthMethods() {
+        List<AuthMethodEntity> authMethods = authMethodRepository.findAllAuthMethods();
+        List<AuthMethod> responseList = new ArrayList<>();
+        if (authMethods == null || authMethods.isEmpty()) {
+            throw new IllegalStateException("No authentication method is configured in Next Step server.");
+        }
+        GetAuthMethodsResponse response = new GetAuthMethodsResponse();
+        for (AuthMethodEntity authMethod : authMethods) {
+            responseList.add(authMethod.getAuthMethod());
+        }
+        response.setAuthMethods(responseList);
+        return new Response<>(Response.Status.OK, response);
+    }
+
+    /**
+     * Get all enabled authentication methods for given user.
+     *
+     * @param userId User ID
+     * @return List of enabled authentication methods for given user wrapped in GetAuthMethodResponse.
+     */
+    @RequestMapping(value = "/user/{userId}/auth-method", method = RequestMethod.GET)
+    public @ResponseBody
+    Response<GetAuthMethodsResponse> getAuthMethodsEnabledForUser(@PathVariable String userId) {
+        List<AuthMethod> authMethods = userPrefsService.listAuthMethodsEnabledForUser(userId);
+        GetAuthMethodsResponse response = new GetAuthMethodsResponse();
+        response.setAuthMethods(authMethods);
+        return new Response<>(Response.Status.OK, response);
+    }
+
+    /**
+     * Enable an authentication method for given user.
+     *
+     * @param userId     User ID
+     * @param authMethod Authentication method
+     * @return List of enabled authentication methods for given user wrapped in GetAuthMethodResponse.
+     */
+    @RequestMapping(value = "/user/{userId}/auth-method/{authMethod}", method = RequestMethod.POST)
```

Comment:
Please favor request object instead of path attributes (both userId and authMethod).